homeassistant/components/sensor/wunderground.py

```diff
@@ -680,46 +719,44 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._cfg_expand("value", STATE_UNKNOWN)
+        return self._state
 
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        attrs = self._cfg_expand("device_state_attributes", {})
-        for (attr, callback) in attrs.items():
-            try:
-                attrs[attr] = callback(self.rest)
-            except TypeError:
-                attrs[attr] = callback
-            except (KeyError, IndexError) as err:
-                _LOGGER.warning("Failed to parse response from WU API: %s",
-                                err)
-
-        attrs[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
-        attrs[ATTR_FRIENDLY_NAME] = self._cfg_expand("friendly_name")
-        return attrs
+        return self._attributes
 
     @property
     def icon(self):
         """Return icon."""
-        return self._cfg_expand("icon", super().icon)
+        return self._icon
 
     @property
     def entity_picture(self):
         """Return the entity picture."""
-        url = self._cfg_expand("entity_picture")
-        if isinstance(url, str):
-            return re.sub(r'^http://', 'https://', url, flags=re.IGNORECASE)
+        return self._entity_picture
 
     @property
     def unit_of_measurement(self):
         """Return the units of measurement."""
-        return self._cfg_expand("unit_of_measurement")
+        return self._unit_of_measurement
 
     def update(self):
         """Update current conditions."""
         self.rest.update()
 
+        if not self.rest.data:
+            # no data, return
+            return
+
+        self._update_state()
+        self._update_attrs()
+        self._icon = self._cfg_expand("icon", super().icon)
+        url = self._cfg_expand("entity_picture")
+        if isinstance(url, str):
+            self._entity_picture = re.sub(r'^http://', 'https://',
```

Comment:
This was a problem with the original code, but why are we rewriting this URL? We should fix the protocol where the URL is coming from.

Reply:
It is workaround to avoid "Mixed Content" browser warnings from PR https://github.com/home-assistant/home-assistant/pull/4544
Unfortunately, WUnderground API is sending icons URL with HTTP protocol regardless of query protocol.